cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcer.java

```diff
@@ -60,6 +63,12 @@ public void enforce(EntityId entity, Principal principal, Action action) throws
   }
 
   private void doEnforce(EntityId entity, Principal principal, Set<Action> actions) throws Exception {
+    if (entity instanceof NamespacedEntityId &&
```

Comment:
What's the point of comparing the acting principal to the current UGI? Why bypass auth check in that case? Can you add a comment, since its not clear to me.